internal/resources/cloudprovider/resource_azure_credential.go

```diff
@@ -0,0 +1,350 @@
+package cloudprovider
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common"
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common/cloudproviderapi"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	//nolint:gosec
+	resourceAzureCredentialTerraformName = "grafana_cloud_provider_azure_credential"
+	resourceAzureCredentialTerraformID   = common.NewResourceID(common.StringIDField("stack_id"), common.StringIDField("resource_id"))
+)
+
+type TagFilter struct {
+	Key   types.String `tfsdk:"key"`
+	Value types.String `tfsdk:"value"`
+}
+
+type resourceAzureCredentialModel struct {
+	ID                 types.String `tfsdk:"id"`
+	Name               types.String `tfsdk:"name"`
+	TenantID           types.String `tfsdk:"tenant_id"`
+	ClientID           types.String `tfsdk:"client_id"`
+	StackID            types.String `tfsdk:"stack_id"`
+	ClientSecret       types.String `tfsdk:"client_secret"`
+	ResourceID         types.String `tfsdk:"resource_id"`
+	ResourceTagFilters []TagFilter  `tfsdk:"resource_tag_filter"`
```

Comment:
I've recently found out that if we use Go slice types directly here, instead of the Terraform type e.g. [List](https://developer.hashicorp.com/terraform/plugin/framework/handling-data/types/list), you can get errors when a value is planned by Terraform with a state other than the normal state e.g. `Unknown`. See https://github.com/grafana/cloud-onboarding/issues/8084